Sources/JavaScriptKit/FundamentalObjects/JSClosure.swift

```diff
@@ -0,0 +1,140 @@
+import _CJavaScriptKit
+
+fileprivate var sharedFunctions: [JavaScriptHostFuncRef: ([JSValue]) -> JSValue] = [:]
+
+/// `JSOneshotClosure` is a JavaScript function that can be called only once.
+public class JSOneshotClosure: JSFunction {
+    private var hostFuncRef: JavaScriptHostFuncRef = 0
+
+    public init(_ body: @escaping ([JSValue]) -> JSValue) {
+        // 1. Fill `id` as zero at first to access `self` to get `ObjectIdentifier`.
+        super.init(id: 0)
+        let objectId = ObjectIdentifier(self)
+        let funcRef = JavaScriptHostFuncRef(bitPattern: Int32(objectId.hashValue))
+        // 2. Retain the given body in static storage by `funcRef`.
+        sharedFunctions[funcRef] = body
+        // 3. Create a new JavaScript function which calls the given Swift function.
+        var objectRef: JavaScriptObjectRef = 0
+        _create_function(funcRef, &objectRef)
+
+        hostFuncRef = funcRef
+        id = objectRef
+    }
+
+    public override func callAsFunction(this: JSObject? = nil, arguments: [ConvertibleToJSValue]) -> JSValue {
+        defer { release() }
+        return super.callAsFunction(this: this, arguments: arguments)
+    }
+
+    /// Release this function resource.
+    /// After calling `release`, calling this function from JavaScript will fail.
+    public func release() {
+        sharedFunctions[hostFuncRef] = nil
+    }
+}
+
+/// `JSClosure` represents a JavaScript function the body of which is written in Swift.
+/// This type can be passed as a callback handler to JavaScript functions.
+/// Note that the lifetime of `JSClosure` should be managed by users manually
+/// due to GC boundary between Swift and JavaScript.
+/// For further discussion, see also [swiftwasm/JavaScriptKit #33](https://github.com/swiftwasm/JavaScriptKit/pull/33)
+///
+/// e.g.
+/// ```swift
+/// let eventListenter = JSClosure { _ in
+///     ...
+///     return JSValue.undefined
+/// }
+///
+/// button.addEventListener!("click", JSValue.function(eventListenter))
+/// ...
+/// button.removeEventListener!("click", JSValue.function(eventListenter))
+/// eventListenter.release()
+/// ```
+///
+public class JSClosure: JSOneshotClosure {
+    
+    var isReleased: Bool = false
+
+    @available(*, deprecated, message: "This initializer will be removed in the next minor version update. Please use `init(_ body: @escaping ([JSValue]) -> JSValue)`")
+    @_disfavoredOverload
+    public init(_ body: @escaping ([JSValue]) -> ()) {
+        super.init({
+            body($0)
+            return .undefined
+        })
+    }
```

Comment:
The issue is with Swift typechecker which can't treat the currently available set of `JSClosure.init` overloads unambiguously without adding an explicit function signature at the place of use ([example one](https://github.com/TokamakUI/Tokamak/blob/a631d181e69e2e72d898b0b778356e61fb03e93b/Sources/TokamakDOM/App/ScenePhaseObserver.swift#L24),  [example two](https://github.com/TokamakUI/Tokamak/blob/3451d9ea12d28f060aa9fa84547e4fdacfc63a8d/Sources/TokamakTestRenderer/TestRenderer.swift#L27)). That defeats the purpose of the overload. I think we'd rather require users to add `return .undefined` explicitly (which can be easily copy-pasted where needed) instead of adding function signatures explicitly (which are almost always ad hoc).